anvil/src/input_handler.rs

```diff
@@ -819,3 +819,15 @@ fn process_keyboard_shortcut(modifiers: ModifiersState, keysym: Keysym) -> Optio
         None
     }
 }
+
+fn detect_terminal() -> String {
+    use std::fs::read_link;
+
+    const SYMLINK: &str = "/usr/bin/x-terminal-emulator";
+
+    if let Ok(found) = read_link(SYMLINK) {
+        return read_link(&found).unwrap_or(found).to_string_lossy().to_string();
+    }
+
+    "/usr/bin/gnome-terminal".to_string()
```

Comment:
Falling back to gnome-terminal is also a no-go.

Many people try anvil on a separate tty, while their main session is still running. If gnome-terminal is already running on the main session, this will only start another window there, not in anvil. Also I would like to avoid having a dependency on a specific desktop environment in anvil.

Anvil is meant as a demo of all of smithays features, not as a full-blown desktop environment. As such relying on test programs like `weston-terminal` (much like X11 examples used to use `xterm`) is perfectly fine imo.